ratelimits/limiter.go

```diff
@@ -265,61 +251,62 @@ func (l *Limiter) Refund(ctx context.Context, name Name, id string, cost int64)
 	// Remove cancellation from the request context so that transactions are not
 	// interrupted by a client disconnect.
 	ctx = context.WithoutCancel(ctx)
-	tat, err := l.source.Get(ctx, bucketKey(name, id))
+	tat, err := l.source.Get(ctx, bucket.key)
 	if err != nil {
 		return nil, err
 	}
-	d := maybeRefund(l.clk, limit, tat, cost)
+	d := maybeRefund(l.clk, limit, tat, bucket.cost)
 	if !d.Allowed {
 		// The bucket is already at maximum capacity.
 		return d, nil
 	}
-	return d, l.source.Set(ctx, bucketKey(name, id), d.newTAT)
-
+	return d, l.source.Set(ctx, bucket.key, d.newTAT)
 }
 
-// Reset resets the specified bucket.
-func (l *Limiter) Reset(ctx context.Context, name Name, id string) error {
+// Reset resets the specified bucket to its maximum capacity. The new bucket
+// state is persisted to the underlying datastore before returning.
+func (l *Limiter) Reset(ctx context.Context, bucket Bucket) error {
```

Comment:
This method is only ever used in `//ratelimits/limiter_test.go`. Would you mind moving it to that test file instead?

Reply:
Yes, this is currently utilized only in test. However, I believe it would be pretty unexpected and non-idiomatic to declare a struct method inside of a test file. I'm going to keep this here for the time being.